snap/README.md

```diff
@@ -114,35 +162,335 @@ Or by using the systemd unit name and `journalctl`:
+sudo snap set edgexfoundry env.security-proxy.public-key="$(cat public.pem)"
+```
+
+The second step is then to generate a token using the user ID which you specified:
+
+```bash
+$ edgexfoundry.secrets-config proxy jwt --algorithm ES256 --private_key private.pem --id USER_ID --expiration=1h
+TOKEN="copy manually from output of secrets-config command"
+```
+
+Alternatively , you can generate the token on a different device using a bash script:
+
+```bash
+header='{
+    "alg": "ES256",
+    "typ": "JWT"
+}'
+
+TTL=$((EPOCHSECONDS+3600)) 
+
+payload='{
+    "iss":"USER_ID",
+    "iat":'$EPOCHSECONDS', 
+    "nbf":'$EPOCHSECONDS',
+    "exp":'$TTL' 
+}'
+
+JWT_HEADER=`echo -n $header | openssl base64 -e -A | sed s/\+/-/ | sed -E s/=+$//`
+JWT_PAYLOAD=`echo -n $payload | openssl base64 -e -A | sed s/\+/-/ | sed -E s/=+$//`
+JWT_SIGNATURE=`echo -n "$JWT_HEADER.$JWT_PAYLOAD" | openssl dgst -sha256 -binary -sign private.pem  | openssl asn1parse -inform DER  -offset 2 | grep -o "[0-9A-F]\+$" | tr -d '\n' | xxd -r -p | base64 -w0 | tr -d '=' | tr '+/' '-_'`
+TOKEN=$JWT_HEADER.$JWT_PAYLOAD.$JWT_SIGNATURE
+```
+
+The resulting JWT token must be included
+via an HTTP `Authorization: Bearer <access-token>` header on any REST calls used to access EdgeX services via the API Gateway. 
+
+Example:
+
+```bash
+$ curl -k -X GET https://localhost:8443/coredata/api/v1/ping? -H "Authorization: Bearer $TOKEN"
+```
+
+Additional users can be added by repeatedly calling the secrets-config command as above. Only one user can however be set at any time when using the snap configure hook, so
+the current user must first be removed by setting the snap configuration settings to an empty string, before setting the values again:
+
+```bash
+$ sudo snap set edgexfoundry env.security-proxy.user=""
+$ sudo snap set edgexfoundry env.security-proxy.public-key=""
+$ sudo snap set edgexfoundry env.security-proxy.user=user02,USER_ID2,ES256
+$ sudo snap set edgexfoundry env.security-proxy.public-key="$(cat public.pem)"
+```
+
+##### OAuth 2.0
+
+OAuth2 is implemented in Kong using the [OAuth 2.0 plugin](https://docs.konghq.com/hub/kong-inc/oauth2/)
+
+The OAuth2 implementation uses the [Client Credentials flow](https://tools.ietf.org/html/rfc6749#section-4.4). In that, the token is generated in two steps
+
+1. The user is created with associated client_id and client_secret values, which can either be specified or generated by the plugin.
+
+2. The client_id and client_secret values are then used to authorize with the authorization server (the Kong plugin) and request an access token from the token endpoint.
+
+The secrets-config command supports this OAuth2 workflow. However, there are some limitations in the EdgeX implementation. 
+
+- The authentication method (jwt or oauth2) needs to be set before EdgeX is installed. During initialization it will set up Kong with the correct authentication plugins and that can only be done once.
```

Comment:
didn't oauth2 get dropped?

Reply:
yes, it did. I removed the oauth2 section